endpoints/api/namespacequota.py

```diff
@@ -75,252 +74,322 @@ class OrganizationQuota(ApiResource):
+            new_type = quota_limit_data["type"]
+            model.namespacequota.update_namespace_quota_limit_type(quota_limit, new_type)
+        if "threshold_percent" in quota_limit_data:
+            new_threshold = quota_limit_data["threshold_percent"]
+            model.namespacequota.update_namespace_quota_limit_threshold(quota_limit, new_threshold)
 
-        if quota is None:
-            msg = "quota limit does not exist"
-            raise request_error(message=msg)
+        return quota_view(quota)
+
+    @nickname("deleteOrganizationQuotaLimit")
+    def delete(self, orgname, quota_id, limit_id):
+        orgperm = AdministerOrganizationPermission(orgname)
+
+        if not features.SUPER_USERS or not SuperUserPermission().can():
+            if not orgperm.can():
+                raise Unauthorized()
+
+        org = model.organization.get_organization(orgname)
+        quota = get_quota(org, quota_id)
+        quota_limit = model.namespacequota.get_namespace_quota_limit(quota, limit_id)
+        if quota_limit is None:
+            raise NotFound()
 
         try:
-            model.namespacequota.change_namespace_quota_limit(
-                namespace,
-                quota_limit_data["percent_of_limit"],
-                quota_limit_data["quota_type_id"],
-                quota_limit_data["quota_limit_id"],
-            )
-            return "Updated", 201
+            model.namespacequota.delete_namespace_quota_limit(quota_limit)
+            return "", 204
         except model.DataModelException as ex:
             raise request_error(exception=ex)
 
-    @nickname("deleteOrganizationQuotaLimit")
-    def delete(self, namespace):
-
-        superperm = SuperUserPermission()
 
-        if not superperm.can():
-            raise Unauthorized()
+@resource("/v1/user/quota")
+@show_if(features.QUOTA_MANAGEMENT)
+class UserQuotaList(ApiResource):
+    @require_user_admin
+    @nickname("listUserQuota")
+    def get(self):
+        parent = get_authenticated_user()
 
-        quota_limit_id = request.args.get("quota_limit_id", None)
-        if quota_limit_id is None:
-            msg = "Bad request to delete quota limit. Missing quota limit identifier."
-            raise request_error(message=msg)
+        user_quotas = model.namespacequota.get_namespace_quota_list(parent)
 
-        quota = model.namespacequota.get_namespace_limit_from_id(namespace, quota_limit_id)
+        return [quota_view(quota) for quota in user_quotas]
 
-        if quota is None:
-            msg = "quota does not exist"
-            raise request_error(message=msg)
 
-        try:
-            success = model.namespacequota.delete_namespace_quota_limit(namespace, quota_limit_id)
-            if success == 1:
-                return "Deleted", 201
+@resource("/v1/user/quota/<quota_id>")
+@show_if(features.QUOTA_MANAGEMENT)
+class UserQuota(ApiResource):
+    @require_user_admin
+    @nickname("getUserQuota")
+    def get(self, quota_id):
```

Comment:
this becomes a duplicate function with the get in the class above because there is only 1 quota per namespace.

Reply:
This one is for the user specifically, whereas the other classes in this module are for organizations.